### PR TITLE
8315965: Open source various AWT applet tests

### PR DESCRIPTION
**Pull request by azvegint**
Open sourcing few applet tests:

java/awt/ScrollPane/ScrollPaneTest.java
java/awt/TextArea/Length.java
java/awt/Window/WindowOwner.java
java/awt/font/Rotate/RotateTest3.java

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8315965](https://bugs.openjdk.org/browse/JDK-8315965): Open source various AWT applet tests (**Bug** - P4)

### Reviewers
 * [Harshitha Onkar](https://openjdk.org/census#honkar) (@honkar-jdk - Committer) ⚠️ Review applies to [7089abe6](https://git.openjdk.org/jdk/pull/15754/files/7089abe63e5da2cc4e94c64dd7aa882ebc85716a)
 * [Prasanta Sadhukhan](https://openjdk.org/census#psadhukhan) (@prsadhuk - **Reviewer**) ⚠️ Review applies to [9e537cc8](https://git.openjdk.org/jdk/pull/15754/files/9e537cc86e2ee3ed7272420e2ccb695ee7ec9a3e)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15754/head:pull/15754` \
`$ git checkout pull/15754`

Update a local copy of the PR: \
`$ git checkout pull/15754` \
`$ git pull https://git.openjdk.org/jdk.git pull/15754/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15754`

View PR using the GUI difftool: \
`$ git pr show -t 15754`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15754.diff">https://git.openjdk.org/jdk/pull/15754.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15754#issuecomment-1720097575)

### CODE REVIEW

File: test/jdk/java/awt/TextArea/Length.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ * questions.
+ */
+
+import java.awt.EventQueue;
+import java.awt.TextArea;
+
+/*
+ * @test
+ * @bug 4120876
+ * @key headful
```
Comment (honkar-jdk) [inline test/jdk/java/awt/TextArea/Length.java:30]: Since we are not displaying anything, can this test be made headless?

Comment (azvegint) [reply test/jdk/java/awt/TextArea/Length.java:30]: No, `TextArea` a heavyweight component and cannot be created in headless environment.

https://github.com/openjdk/jdk/blob/master/src/java.desktop/share/classes/java/awt/TextComponent.java#L150

File: test/jdk/java/awt/font/Rotate/RotateTest3.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        frame.setVisible(true);
+    }
+
+    static public class Java2DView extends Component {
+
+        public void paint(Graphics g){
+            Graphics2D g2d = (Graphics2D) g;
+            Dimension d = this.getSize();
+            g.setColor(this.getBackground());
+            g.fillRect(0,0,d.width,d.height);
```
Comment (honkar-jdk) [inline test/jdk/java/awt/font/Rotate/RotateTest3.java:93]:
```suggestion
            g.fillRect(0, 0, d.width, d.height);
```

File: test/jdk/java/awt/TextArea/Length.java (hunk 1 above)
Comment (honkar-jdk) [reply test/jdk/java/awt/TextArea/Length.java:30]: Sounds good!

**Review by honkar-jdk — APPROVED** [review]
LGTM

File: test/jdk/java/awt/font/Rotate/RotateTest3.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+            });
+        }
+    }
+
+    private static void initAndShowGui() {
+        RotateTest3 panel = new RotateTest3();
+
+        frame = new JFrame("RotateTest3");
+        frame.addWindowListener(new WindowAdapter(){
+            public void windowClosing(WindowEvent e) {System.exit(0);}
```
Comment (prsadhuk) [inline test/jdk/java/awt/font/Rotate/RotateTest3.java:78]: I guess we should remove System.exit from the test as it may hamper CI

**Review by prsadhuk — APPROVED** [review]

**Pull request closed without merging**